### ISSUE TITLE
Code editor settings can only be enabled, not disabled in Notebook

### ISSUE THREAD

**Issue by krassowski** [jupyterlab/jupyterlab#15246]
## Description

Checking a setting like `Auto Closing Brackets`, `Highlight the active line`, `Highlight white spaces` and others in notebook enables it but unchecking does not disable it afterwards.

## Reproduce

1. Open a notebook and Settings → Notebook 
2. Enable `Highlight the active line`
4. Click on the cell, see active line highlighted
5. Disable `Highlight the active line`
6. Click on the same cell, see active line remains hihglighted
7. Create a new cell, see that the settings are ok in newly created cells

![many-settings-do-not-deactivate](https://github.com/jupyterlab/jupyterlab/assets/5832902/e234630c-694b-4956-87ed-a86dabcf92f9)

## Expected behavior

Settings get reflected in the existing cells

## Context

- Browser and version: Chrome
- JupyterLab version: 4.0.6 and 4.1.0a1

**Comment by JasonWeill** [comment]
Does the "disabled" setting show up as enabled in the settings editor after you reload your JupyterLab session?

**Comment by krassowski** [comment]
Yes. The issue is that existing cells do not ever get extensions de-registered, not that the settings system is not saving. I had included step  `6. Create a new cell, see that the settings are ok in newly created cells` to hint at it.

**Comment by firai** [comment]
I'm not sure if it's the same issue, but I'm noticing that changes to CodeMirror settings only update CM instances when the new value is different from the default. When they are changed back to the default value, the previous value sticks.

**Comment by krassowski** [comment]
Still an issue as of 4.6.0
